js/ui/codeView.js

```diff
@@ -220,13 +220,16 @@ var WindowTrackingButton = new Lang.Class({
                                                       this._updatePosition.bind(this));
         this._sizeChangedId = this.window.connect('size-changed',
                                                   this._updatePosition.bind(this));
-
         this._windowsRestackedId = Main.overview.connect('windows-restacked',
                                                          this._showIfWindowVisible.bind(this));
         this._overviewHidingId = Main.overview.connect('hiding',
                                                        this._showIfWindowVisible.bind(this));
         this._overviewShowingId = Main.overview.connect('showing',
                                                         this._hide.bind(this));
+        this._screenShieldActiveChangedId = Main.screenShield.actor.connect('notify::visible',
```

Comment:
I don't think this is the best approach; instead I would track the `sessionMode`, look at the `isLocked` property, and disable this feature when that's true.